docs/USER_GUIDE.md

```diff
@@ -14,7 +35,43 @@ The module names follow the typical hierarchical module
 naming scheme with top-level names like `Control`, `Data`, `System`
 and so on.
 
-## Temporary limitations
+
+## (2) Major design decisions that set standards
+
+If you aren't familiar with these design decisions, large parts of `linear-base`
+might be confusing or you might write code that goes against the grain of
+how everything is set up.
+
+### Interfacing linear and non-linear code
+
+Throughout linear code, you will need to interface between non-linear
+and linear code. All the tools you need to do this are in
+[`Data.Unrestricted`] and are typically re-exported by
+[`Prelude.Linear`].
+
+This is basically done through type classes. Types that can be used in
+a non-linear way even when they are bound in a linear function have
+instances of one or more of these things: `Consumable, Dupable,
+Moveable`.
+
+### `f :: X -> (SomeType #-> Ur b) #-> Ur b` functions
+
+This function limits the **scope** of using `SomeType` by taking
+a scope function of type `(SomeType #-> Ur b)`
+as its second argument and using it with a value of type `SomeType` to
+produce an `Ur b`.
+
+The `SomeType` cannot escape by having it be inside the type `b` 
+in some way. This is because the `SomeType` is bound linearly in the scope
+function and `Ur` holds its value with a non-linear constructor
+arrow.
+
+Now, if `f` is the only function that can make a `SomeType`,
+then we have an API that completely controls the creation-to-deletion
+lifetime (i.e, the scope) of `SomeType` and ensures it is used linearly.
```

Comment:
I like to call the function argument the _scope_ probably this piece of terminology should be explained here, so that we can refer to scopes in the documentation.

Reply:
Agreed :+1: